src/server/auth-passport.js

```diff
@@ -74,37 +74,38 @@ export function setupAuth0Passport() {
 
 export function setupLocalAuthPassport() {
   const strategy = new LocalStrategy({
-    usernameField: 'email'
-  }, wrap(async (username, password, done) => {
-    console.log('LOGIN', username, password)
-    const existingUser = await User.filter({ email: username })
+    usernameField: 'email',
+    passReqToCallback: true
+  }, wrap(async (req, username, password, done) => {
+    const lowerCaseEmail = username.toLowerCase()
+    const existingUser = await User.filter({ email: lowerCaseEmail })
     if (existingUser.length > 0) {
       const pwFieldSplit = existingUser[0].auth0_id.split('|')
       const hashed = {
-        salt: pwFieldSplit[0],
-        hash: pwFieldSplit[1]
+        salt: pwFieldSplit[1],
+        hash: pwFieldSplit[2]
       }
       AuthHasher.verify(
         password, hashed,
         (err, verified) => (verified
-                            ? done(null, existingUser[0])
-                            : done('Validation failed', null))
+          ? done(null, existingUser[0])
+          : done(null, false))
       )
     } else {
       // create the user
-      AuthHasher.hash(password, async function(err, hashed) {
-        const passwordToSave = `${hashed.salt}|${hashed.hash}`
-          //.salt and .hash
+      AuthHasher.hash(password, async function (err, hashed) {
```

Comment:
here we need to validate an invite code or org uuid that is passed in to make sure they have been invited to create an account first.

Reply:
I think we also want to validate the passwords are the same here.  Just because we validate on the client, we also have to validate on the server.

Reply:
> here we need to validate an invite code or org uuid that is passed in to make sure they have been invited to create an account first.

Do we need to also do this when using Auth0? Right now, when a texter signs up using either local login or Auth0, the following message is displayed:
![image](https://user-images.githubusercontent.com/17498598/52885583-90d66a00-3126-11e9-8676-04ca109c2933.png)

Reply:
If we can then yes, that would actually be good for security. But if it's tricky then you don't need to do it for this PR